Library/include/CSP/Systems/Sequence/SequenceSystem.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+
+#include "CSP/CSPCommon.h"
+#include "CSP/Common/String.h"
+#include "CSP/Systems/Sequence/Sequence.h"
+#include "CSP/Systems/SystemBase.h"
+
+namespace csp::web
+{
+
+class WebClient;
+
+} // namespace csp::web
+
+
+namespace csp::memory
+{
+
+CSP_START_IGNORE
+template <typename T> void Delete(T* Ptr);
+CSP_END_IGNORE
+
+} // namespace csp::memory
+
+namespace csp::systems
+{
+/// @ingroup Sequence System
+/// @brief Public facing system that allows the management of groupings of items in a space.
+class CSP_API SequenceSystem : public SystemBase
+{
+	CSP_START_IGNORE
+	/** @cond DO_NOT_DOCUMENT */
+	friend class SystemsManager;
+	friend void csp::memory::Delete<SequenceSystem>(SequenceSystem* Ptr);
+	/** @endcond */
+	CSP_END_IGNORE
+public:
+	/// @brief
+	/// @param SequenceKey csp::common::String : The unique grouping name. The suggested convention is: Type:[Id]
+	/// @param ReferenceType csp::common::String : The type of reference (GroupId, SpaceId etc.)
+	/// @param ReferenceId csp::common::String : The id of the reference
+	/// @param Items csp::common::Array<csp::common::String> : An ordered array of members
+	/// @param Callback SequenceResultCallback : callback to call when a response is received
+	void CreateSequence(const csp::common::String& SequenceKey,
```

Comment:
Is it worth adding a note for the API consumer that reusing a sequence key would not be considered an error but will overwrite that sequence?

Reply:
Good idea!